### PR TITLE
Feat/v2 to v3 translation updates

### PR DESCRIPTION
**Pull request by JieXiong9119**
#### What does this PR do?

This PR includes follow-up updates on the version translation tool developed in #348. The updated v2_to_v3.xsl file should bring in all the breaking changes from v2 to v3, for transforming BuildingSync v2 xml instances to v3 (via any xslt processor).

#### What are the relevant PRs?

Breaking changes that are accounted for in #348: 
- #335 
- #341 
 
Breaking changes added into xsl on this PR:
- #354 
- #353 
- #352 
- #351 
- #345 
- #342 
- #334 
- #331 
- #329

### CODE REVIEW

File: translation/xsltproc.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+# Path to the output xml file
+output_path = "new_xml.xml"
+
+xml_origin = ET.parse(xml_path)
+xslt = ET.parse(xsl_path)
+transform = ET.XSLT(xslt)
+xml_new = transform(xml_origin)
+
+with open(output_path, "wb") as f:
+    f.write(ET.tostring(xml_new, pretty_print=True))
```
Comment (macintoshpie) [inline translation/xsltproc.py:18]: Could you update this to use the newer method we found for including the declaration?

File: translation/v2_to_v3.xsl [hunk 2]
```diff
@@ -52,4 +56,94 @@ Version: 0.1.0
...
+  <xsl:template match="auc:FutureOtherECMs">
+    <xsl:element name="auc:Uncategorized">
+      <xsl:apply-templates/>
+    </xsl:element>
+  </xsl:template>
+
+  <!-- Move enumeration "Add or replace cooling tower" from auc:OtherHVAC/auc:MeasureName to auc:ChillerPlantImprovements/auc:MeasureName -->
+  <!-- Move enumeration "Install or upgrade master venting" from auc:OtherHVAC/auc:MeasureName to auc:ChilledWaterHotWaterAndSteamDistributionSystems/auc:MeasureName -->
+  <!-- Move enumeration "Separate SHW from heating" from auc:ChilledWaterHotWaterAndSteamDistributionSystems/auc:MeasureName to auc:ServiceHotWaterSystems/auc:MeasureName -->
+
```
Comment (macintoshpie) [inline translation/v2_to_v3.xsl:148]: Are these todos?

**Review by macintoshpie — COMMENTED** [review]
Awesome! Could you add examples of these in the buildingsync test file so the transformations are tested? Once these comments get wrapped up this should be good to go

File: translation/v2_to_v3.xsl (hunk 2 above)
Comment (JieXiong9119) [reply translation/v2_to_v3.xsl]: Yes, and I'm still struggling with the implementation. Any good idea?

File: translation/xsltproc.py (hunk 1 above)
Comment (JieXiong9119) [reply translation/xsltproc.py]: Updated!

**Comment by JieXiong9119** [comment]
> Awesome! Could you add examples of these in the buildingsync test file so the transformations are tested? Once these comments get wrapped up this should be good to go

I just created a test.xml file in the subfolder. Could you take a look?

**Review by macintoshpie — CHANGES_REQUESTED** [review]
Hey Jie, I moved your example file into the spec/data/v2_for_translation.xml file b/c this is the one that actually gets tested.

After doing that there are some failures, it looks like we are still missing some translations. Could you add them to the translator?

Screen Shot 2021-06-11 at 11 46 33 AM


Screen Shot 2021-06-11 at 11 46 42 AM

**Review by macintoshpie — APPROVED** [review]
Great! 💥

**Pull request merged**